.github/workflows/docs.yaml

```diff
@@ -4,50 +4,71 @@ on:
   push:
     branches:
       - master
-      - v*
   pull_request:
 
 jobs:
-  test-docs:
-    name: Test Docs
+  test-or-deploy:
+    name: ${{ github.event_name == 'push' && 'Deploy' || 'Test' }}
     runs-on: ubuntu-latest
+    container: debian:bookworm
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - name: Install Dependencies
+        run: |
+          apt-get update
+          apt-get -y install \
+            build-essential \
+            ca-certificates \
+            flatbuffers-compiler-dev \
+            g++-10 \
+            gcc-10 \
+            git \
+            gnupg2 \
+            jq \
+            libflatbuffers-dev \
+            libfmt-dev \
+            libpcap-dev tcpdump \
+            libsimdjson-dev \
+            libspdlog-dev \
+            libssl-dev \
+            libunwind-dev \
+            libyaml-cpp-dev \
+            libxxhash-dev \
+            lsb-release \
+            ninja-build \
+            pkg-config \
+            python3-dev \
+            python3-pip \
+            python3-venv \
+            wget \
+            yarnpkg
+          python3 -m pip install --upgrade pip
+          python3 -m pip install --upgrade cmake
+      - name: Checkout
+        uses: actions/checkout@v3
         with:
-          node-version: 16.x
-          cache-dependency-path: docs/yarn.lock
-          cache: yarn
-      - name: Install dependencies
-        working-directory: docs
-        run: yarn install --frozen-lockfile
-      - name: Test build website
-        working-directory: docs
-        run: yarn build
-
-  deploy-docs:
-    name: Deploy Docs
-    if: ${{ github.event_name == 'push' }}
-    runs-on: ubuntu-latest
-    needs:
-      - test-docs
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
-        with:
-          node-version: 16.x
-          cache-dependency-path: docs/yarn.lock
-          cache: yarn
-      - name: Install dependencies
-        working-directory: docs
-        run: yarn install --frozen-lockfile
-      - name: Test build website
-        working-directory: docs
-        run: yarn build
```

Comment:
I find this way more maintainable than going into CMake land. Is there no way around doing this with `yarn`?

Reply:
Nope. We amalgamate the changelog using CMake as part of the build already, so unless we wanna exclude the files from the last release by using the copy in the source tree root that we create with every release we'll have to go through it.

I don't think maintainability needs to be a concern here, it's pretty straightforward if you ignore the yet to be simplified installation of dependencies above.

Reply:
Take a look now, it's actually much simpler now that we re-use the output of the existing changelog job.